core/bindings.rs

```diff
@@ -1,144 +1,26 @@
 use log::debug;
 use once_cell::sync::Lazy;
-use serde::Deserialize;
-use serde::Serialize;
-use serde_v8::to_v8;
-use std::cell::RefCell;
-use std::collections::HashMap;
 use std::option::Option;
 use std::os::raw::c_void;
-use url::Url;
-use v8::HandleScope;
-use v8::Local;
 use v8::MapFnTo;
-use v8::SharedArrayBuffer;
-use v8::ValueDeserializerHelper;
-use v8::ValueSerializerHelper;
 
 pub static EXTERNAL_REFERENCES: Lazy<v8::ExternalReferences> =
   Lazy::new(|| {
-    v8::ExternalReferences::new(&[
-      v8::ExternalReference {
-        function: ref_op.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: unref_op.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_macrotask_callback.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_nexttick_callback.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_promise_reject_callback.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_uncaught_exception_callback.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: run_microtasks.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: has_tick_scheduled.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_has_tick_scheduled.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: eval_context.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: queue_microtask.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: create_host_object.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: encode.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: decode.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: serialize.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: deserialize.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: get_promise_details.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: get_proxy_details.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: is_proxy.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: memory_usage.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: call_console.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: set_wasm_streaming_callback.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: abort_wasm_streaming.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: destructure_error.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: terminate.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: apply_source_map.map_fn_to(),
-      },
-      v8::ExternalReference {
-        function: op_names.map_fn_to(),
-      },
-    ])
+    v8::ExternalReferences::new(&[v8::ExternalReference {
```

Comment:
Cool that you limited number of external references, that should simplify snapshotting a bit. Ideally we'd remove all of them, but that's outside of scope for this PR.

Reply:
In the future I would like to make ops-snapshottable, avoiding the teardown/setup we currently have and allowing code to hold long lived references avoid the indirection and IC costs of lookups.

Reply:
So we'll likely never provide zero external refs, rather than a fixed count it will become variable (per isolate)